enterprise_access/apps/bffs/tests/test_handlers.py

```diff
@@ -0,0 +1,125 @@
+            base_handler.add_error(**malformed_output)
+
+    def test_base_handler_add_warning_serializer(self):
+        base_handler = self.base_handler
+        expected_output = {
+            "developer_message": "Heuristic Expiration",
+            "user_message": "The data received might be out-dated",
+        }
+        # Define kwargs for add_warning
+        arguments = {
+            **expected_output,
+            "status": 113  # Add an attribute that is not explicitly defined in the serializer to verify
+        }
+        base_handler.add_warning(
+            **arguments
+        )
+        self.assertEqual(expected_output, base_handler.context.warnings[0])
+
+    def test_base_handler_add_warning_serializer_is_valid(self):
+        base_handler = self.base_handler
+        malformed_output = {
+            "user_message": "The data received might be out-dated",
+        }
+        with self.assertRaises(ValidationError):
+            base_handler.add_error(**malformed_output)
+
+
+class TestBaseLearnerPortalHandler(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.user = self.mock_user
+        self.context = HandlerContext(self.request)
+        self.base_handler = BaseHandler(self.context)
+        self.base_learner_portal_handler = BaseLearnerPortalHandler(self.base_handler)
+
+    # TODO: Test pertaining to currently stubbed out functions deferred for future tickets
+
+
+class TestDashboardHandler(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.user = self.mock_user
+        self.context = HandlerContext(self.request)
+        self.base_handler = BaseHandler(self.context)
+        self.base_learner_portal_handler = BaseLearnerPortalHandler(self.base_handler)
+        self.dasboard_handler = DashboardHandler(self.base_learner_portal_handler)
+
+    # TODO: Update tests once stubbed our function updated
+    def test_load_and_process(self):
+        dashboard_handler = self.dasboard_handler
+        dashboard_handler.load_and_process()
+        self.assertEqual(
+            dashboard_handler.context.data['enterprise_course_enrollments'],
+            dashboard_handler.get_enterprise_course_enrollments()
```

Comment:
[curious] I believe we may want to hardcode the expected data vs. relying on calling `dashboard_handler.get_enterprise_course_enrollments()` here? Presumably the data in the `context.data['enterprise_course_enrollments']` will always equal `dashboard_handler.get_enterprise_course_enrollments`?